### PR TITLE
Swift3

### PR DESCRIPTION
**Pull request by mtnbarreto**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,12 +1,53 @@
 # Change Log
 All notable changes to this project will be documented in this file.
 
-### master branch
+### [2.0.0-beta.1](https://github.com/xmartlabs/Eureka/releases/tag/2.0.0)
+
+Pull requests associated with this milestone can be found in this [filter](https://github.com/xmartlabs/Eureka/issues?utf8=%E2%9C%93&q=milestone%3A2.0.0%20).
+
+We have made tons of changes in Eureka API to follow the new Swift API design guidelines.
```
Comment (mats-claassen) [inline CHANGELOG.md]: tons of changes `to the` Eureka API

File: Documentation/Eureka 2.0 Migration  Guide.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Eureka 2.0 Migration  Guide
+
+Eureka 2.0.0 includes complete Swift 3 Compatibility and adopts the new [API Design Guidelines](https://swift.org/documentation/api-design-guidelines/). It also includes new validations build-in feature.
```
Comment (mats-claassen) [inline Documentation/Eureka 2.0 Migration  Guide.md]: `It also includes validations as a build-in features`

File: Documentation/Eureka 2.0 Migration  Guide.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+Bring support to swift 3 involves some API updates to follow apple Swift API best practices, we have also changed and deprecated some API.
```
Comment (mats-claassen) [inline Documentation/Eureka 2.0 Migration  Guide.md]: Apple's
deprecated some `API's`

File: Documentation/Eureka 2.0 Migration  Guide.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+|`func presentViewController(_ viewController: VCType!, row: BaseRow, presentingViewController:FormViewController)`|`present(_ viewController: VCType!, row: BaseRow, presentingViewController:FormViewController)`|
+|`func createController() -> VCType?`|`func makeController() -> VCType?`|
+
+
+There are also some breaking changes related with deprecated API:
+
+Removed APIs:
+
+* `PostalAddressRow` and `ImageRow` was deleted.
+*  row has been deleted. You can find it and many other custom rows at EurekaCommunity [organization account](https://github.com/eurekaCommunity).
```
Comment (mats-claassen) [inline Documentation/Eureka 2.0 Migration  Guide.md]: Remove `row has been deleted.  `

Comment (mats-claassen) [inline Documentation/Eureka 2.0 Migration  Guide.md]: You can find them and many other custom rows at EurekaCommunity's

File: Documentation/Eureka 2.0 Migration  Guide.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+`highlightCell` and `unhighlightCell` callbacks were deleted, now we should use `row.isHighlighted` from cell update to check from highlighted status and make UI modification according its value.
+
+In case you want to do something when highligth state switches its value you can set up `onCellHighlightChanged` callback.
```
Comment (mats-claassen) [inline Documentation/Eureka 2.0 Migration  Guide.md]: `highlight`

File: Documentation/Eureka 2.0 Migration  Guide.md [hunk 6]
```diff
@@ -32,12 +35,10 @@ There are also some breaking changes related with deprecated API:
 
 Removed APIs:
 
-* `PostalAddressRow` and `ImageRow` was deleted.
-*  row has been deleted. You can find it and many other custom rows at EurekaCommunity [organization account](https://github.com/eurekaCommunity).
-
-`highlightCell` and `unhighlightCell` callbacks were deleted, now we should use `row.isHighlighted` from cell update to check from highlighted status and make UI modification according its value.
+* `PostalAddressRow` and `ImageRow` was deleted. You can find them and many other custom rows at EurekaCommunity [organization account](https://github.com/eurekaCommunity).
```
Comment (mats-claassen) [inline Documentation/Eureka 2.0 Migration  Guide.md]: `PostalAddressRow` and `ImageRow` **were** deleted

File: README.md [hunk 7]
```diff
@@ -377,7 +377,83 @@ To easily get the selected row/s of a `SelectableSection` there are two methods:
...
+
+Eureka allows us to specify when validation rules should be evaluated. We can do it by setting up `validationOptions` row's property, which can have the following values:
+
+
+* validatesOnChange - Validates whenever a row value changes.
+* validatesOnBlur - (Default value) validates right after the cell resigns first responder. Not applicable for all rows.
+* validatesOnChangeAfterBlurred - Validates whenever the row value changes after it resigns first responder for the first time.
+* validatesOnDemand - We should manually validate the row or form by invoking `validate()` method.
+
+If you want to validate the entire form (all the rows) you can invoke Form `validate()` method.
```
Comment (mats-claassen) [inline README.md]: you can invoke `form.validate()` is more clear

File: Documentation/Eureka 2.0 Migration  Guide.md (hunk 6 above)
Comment (weyert) [inline Documentation/Eureka 2.0 Migration  Guide.md]: Where can I find these rows in EurekaCommunity?

Comment (mats-claassen) [inline Documentation/Eureka 2.0 Migration  Guide.md]: They are currently in progress. We will soon add them there. I believe the ImageRow has also been added to the Examples project.

Comment (weyert) [inline Documentation/Eureka 2.0 Migration  Guide.md]: Yeah, I just noticed that while looking for examples to make my custom rows working again. Swift 3 really is a pain in the neck!

**Pull request merged**